compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -14572,6 +14573,45 @@ private STNode parseXMLNamePatternChain(STNode startToken) {
         return STNodeFactory.createXMLNamePatternChainingNode(startToken, xmlNamePattern, gtToken);
     }
 
+    /**
+     * Parse xml step extend chain.
+     * <p>
+     * <code>
+     * xml-step-extend := .< xml-name-pattern > | [ expression ] | .method-name ( arg-list )
+     * </code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseXMLStepExtend() {
+        List<STNode> xmlStepExtendList = new ArrayList<>();
+        STToken nextToken = peek();
+
+        if (isEndOfXMLStepExtend(nextToken.kind)) {
+            return STNodeFactory.createNodeList(xmlStepExtendList);
+        }
+
+        startContext(ParserRuleContext.XML_STEP_EXTEND);
+        STNode stepExtension;
+        while (!isEndOfXMLStepExtend(nextToken.kind)) {
+            if (nextToken.kind == SyntaxKind.DOT_TOKEN) {
+                STNode dotToken = parseDotToken();
+                STNode funcCallExpression = parseFuncCall(parseIdentifier(ParserRuleContext.IDENTIFIER));
```

Comment:
[function-call-expr](https://ballerina.io/spec/lang/master/#function-call-expr) vs [method-call-expr](https://ballerina.io/spec/lang/master/#method-call-expr): they have slightly different grammar. 
e.g.
-  function call allows qualified identifier. Method call does not allow it.
- method-name allows `special-method-name := map | join | start`. Function reference does not allow it

So I think we better parse `. method-name ( arg-list )` items separately. 
i.e.
```
parseMethodCallExprRhs() {
   parseDotToken()
   parseMethodName()
   parseParenthesizedArgList()
}
```